# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/WSRunObject.h
# Commit: 71670cedaf26
# Full Hash: 71670cedaf26214f3bcb8562a445cf1ef24f4db6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-06-30 09:36:48
# Regressor Bug: 1714914
# File Overlap Count: 1
# Description:
#   Bug 1714914 - Make editor classes treat offset in parent node as uint32_t r=m_kato
#   
#   It should be treated as `uint32_t` since DOM API does so.  However, there are
#   some exceptions:
#   
# ==============================================================================

diff -r f5dcc2d793d6 -r 71670cedaf26 editor/libeditor/WSRunObject.h
--- a/editor/libeditor/WSRunObject.h	Tue Jun 29 22:00:09 2021 +0000
+++ b/editor/libeditor/WSRunObject.h	Wed Jun 30 07:07:28 2021 +0000
@@ -727,7 +727,7 @@
   EditorDOMPointInText GetPreviousCharPointFromPointInText(
       const EditorDOMPointInText& aPoint) const;
 
-  char16_t GetCharAt(dom::Text* aTextNode, int32_t aOffset) const;
+  char16_t GetCharAt(dom::Text* aTextNode, uint32_t aOffset) const;
 
   /**
    * TextFragmentData stores the information of white-space sequence which
@@ -1355,7 +1355,7 @@
   // before {aSplitNode,aSplitOffset} needs to end with nbsp.
   MOZ_CAN_RUN_SCRIPT static nsresult PrepareToSplitAcrossBlocks(
       HTMLEditor& aHTMLEditor, nsCOMPtr<nsINode>* aSplitNode,
-      int32_t* aSplitOffset);
+      uint32_t* aSplitOffset);
 
   /**
    * MergeFirstLineOfRightBlockElementIntoDescendantLeftBlockElement() merges